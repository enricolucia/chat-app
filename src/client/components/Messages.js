import React, { Component } from 'react'
import Message from './Message'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledMessages = styled.div`

`

class Messages extends Component {
  render () {
    const { messages } = this.props
    return (
      <StyledMessages innerRef={ref => { this.messagesWrapper = ref }}>
        {messages.map((message) =>
          <Message
            key={message.id}
            data={message}
            isCountdown={message.countdown} />
          )
        }
      </StyledMessages>
    )
  }
}

Messages.propTypes = {
  messages: PropTypes.array
}

export default Messages
