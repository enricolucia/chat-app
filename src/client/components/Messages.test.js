/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Messages from './Messages'

const defaultProps = {
  messages: [
    {
      id: '0',
      data: {
        countdown: false,
        content: 'Lorem ipsum',
        highlighted: false,
        think: false,
        own: true
      }
    },
    {
      id: '1',
      data: {
        countdown: false,
        content: 'Donor semper',
        highlighted: false,
        think: false,
        own: false
      }
    }
  ]
}

describe('Messages', () => {
  it('renders list of messages', () => {
    const tree = renderer.create(
      <Messages {...defaultProps} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
