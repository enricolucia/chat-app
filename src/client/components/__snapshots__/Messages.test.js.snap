// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages renders list of messages 1`] = `
<div
  className="Messages__StyledMessages-fVsryA hmhJma"
>
  <div
    className="Message__StyledMessage-ktMMqk bhouUx"
    data={
      Object {
        "data": Object {
          "content": "Lorem ipsum",
          "countdown": false,
          "highlighted": false,
          "own": true,
          "think": false,
        },
        "id": "0",
      }
    }
  />
  <div
    className="Message__StyledMessage-ktMMqk bhouUx"
    data={
      Object {
        "data": Object {
          "content": "Donor semper",
          "countdown": false,
          "highlighted": false,
          "own": false,
          "think": false,
        },
        "id": "1",
      }
    }
  />
</div>
`;
